src/producer.rs

```diff
@@ -26,7 +26,8 @@ use util::cstr_to_owned;
 
 /// A native rdkafka-sys topic, used for message production.
 struct NativeTopic {
-    ptr: *mut RDKafkaTopic,
+    client_ptr: *mut RDKafka,
```

Comment:
Including the `client_ptr` here is dangerous, since we might access a pointer that has been freed by another section of the code (it can't happen with `topic_ptr` since it's freed in the `Drop` method).

The right approach is probably to have a real reference or `Arc` to the client. I'll merge the change and then see how I can change that bit. It will probably require a bit of refactoring.